mllib/src/main/scala/org/apache/spark/mllib/clustering/GaussianMixtureModel.scala

```diff
@@ -159,19 +159,25 @@ object GaussianMixtureModel extends Loader[GaussianMixtureModel] {
 
       // Create Parquet data.
       val dataArray = Array.tabulate(weights.length) { i =>
-        Data(weights(i), gaussians(i).mu, gaussians(i).sigma)
+        Row(weights(i), gaussians(i).mu, gaussians(i).sigma)
       }
-      sc.parallelize(dataArray, 1).toDF().write.parquet(Loader.dataPath(path))
+      val dataRDD: RDD[Row] = sc.parallelize(dataArray, 1)
+
+      sqlContext.createDataFrame(dataRDD, schema).write.parquet(Loader.dataPath(path))
     }
+    private val schema = StructType(
+      StructField("weight", DoubleType, nullable = false)::
+      StructField("mu", new VectorUDT, nullable = false)::
+      StructField("sigma", new MatrixUDT, nullable = false)::Nil)
 
     def load(sc: SparkContext, path: String): GaussianMixtureModel = {
       val dataPath = Loader.dataPath(path)
       val sqlContext = new SQLContext(sc)
       val dataFrame = sqlContext.read.parquet(dataPath)
-      val dataArray = dataFrame.select("weight", "mu", "sigma").collect()
```

Comment:
Loader.checkSchema was called to verify the schema after dataframe.select(...). 
Schema verification should be done before  dataframe.select(...),  thats why I changed the order